SMM Crash on first launch: panic: runtime error: index out of range [0] with length 0

I just downloaded the AppImage from your website and tried launching it for the first time. It looks like it correctly identified the installation location of satisfactory and then crashes:

```
[msrd0g@msrd0-tower ~]$ ./SatisfactoryModManager_linux_amd64.AppImage
2024-10-22T15:20:32+02:00 INF SatisfactoryModManager/main.go:55 starting Satisfactory Mod Manager version=3.0.3 commit=eab4298f6fdfd6b14c10d158a34cd3b7807783de date=2024-10-21T18:40:33Z type=appimage
2024-10-22T15:20:32+02:00 INF cli/profiles.go:177 saving profiles path=/home/msrd0g/.local/share/ficsit/profiles.json
2024-10-22T15:20:32+02:00 INF cli/installations.go:102 saving installations path=/home/msrd0g/.local/share/ficsit/installations.json
2024-10-22T15:20:33+02:00 INF ficsitcli/metadata.go:27 failed to find installations error="Legendary not installed"
2024-10-22T15:20:33+02:00 INF ficsitcli/metadata.go:27 failed to find installations error="Heroic not installed"
2024-10-22T15:20:33+02:00 INF ficsitcli/metadata.go:27 failed to find installations error="failed to get snap path: failed to read heroic snap dir: open /home/msrd0g/snap/heroic: no such file or directory"
2024-10-22T15:20:33+02:00 INF ficsitcli/metadata.go:27 failed to find installations error="Epic is not installed in /home/msrd0g/.local/share/Steam/steamapps/common/Among Us"
[...]
2024-10-22T15:20:33+02:00 INF ficsitcli/metadata.go:27 failed to find installations error="Epic is not installed in /tmp"
2024-10-22T15:20:33+02:00 INF ficsitcli/metadata.go:27 failed to find installations error="failed to run lutris -lj: exec: \"flatpak\": executable file not found in $PATH"
2024-10-22T15:20:33+02:00 INF ficsitcli/metadata.go:27 failed to find installations error="Heroic not installed"
2024-10-22T15:20:33+02:00 INF ficsitcli/metadata.go:27 failed to find installations error="steam-flatpak not installed"
2024-10-22T15:20:33+02:00 INF ficsitcli/metadata.go:27 failed to find installations error="steam-snap not installed"
2024-10-22T15:20:33+02:00 INF disk/main.go:58 using local disk path=/home/msrd0g/HDD/SteamLibrary/steamapps/common/Satisfactory
2024-10-22T15:20:33+02:00 INF cli/installations.go:102 saving installations path=/home/msrd0g/.local/share/ficsit/installations.json
2024-10-22T15:20:33+02:00 INF cli/installations.go:102 saving installations path=/home/msrd0g/.local/share/ficsit/installations.json
Overriding existing handler for signal 10. Set JSC_SIGNAL_FOR_GC if you want WebKit to use a different signal
Could not determine the accessibility bus address
panic: runtime error: index out of range [0] with length 0

goroutine 63 [running]:
github.com/satisfactorymodding/SatisfactoryModManager/backend/utils.GetDisplayBounds()
	/home/runner/work/SatisfactoryModManager/SatisfactoryModManager/backend/utils/display.go:22 +0x205
github.com/satisfactorymodding/SatisfactoryModManager/backend/app.(*app).ensureWindowVisible(0xc00047cfb0?)
	/home/runner/work/SatisfactoryModManager/SatisfactoryModManager/backend/app/window.go:41 +0x8c
github.com/satisfactorymodding/SatisfactoryModManager/backend/app.(*app).WatchWindow.func1()
	/home/runner/work/SatisfactoryModManager/SatisfactoryModManager/backend/app/window.go:24 +0x30
created by github.com/satisfactorymodding/SatisfactoryModManager/backend/app.(*app).WatchWindow in goroutine 16
	/home/runner/work/SatisfactoryModManager/SatisfactoryModManager/backend/app/window.go:18 +0xa7
```

These are the files that were created in `~/.local/share/ficsit`:

```
[msrd0g@msrd0-tower ~]$ bat .local/share/ficsit/installations.json
───────┬────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
       │ File: .local/share/ficsit/installations.json
───────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1   │ {
   2   │   "selected_installation": "/home/msrd0g/HDD/SteamLibrary/steamapps/common/Satisfactory",
   3   │   "installations": [
   4   │     {
   5   │       "path": "/home/msrd0g/HDD/SteamLibrary/steamapps/common/Satisfactory",
   6   │       "profile": "Default",
   7   │       "vanilla": false
   8   │     }
   9   │   ],
  10   │   "version": 0
  11   │ }
───────┴────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
[msrd0g@msrd0-tower ~]$ bat .local/share/ficsit/profiles.json
───────┬────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
       │ File: .local/share/ficsit/profiles.json
───────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1   │ {
   2   │   "profiles": {
   3   │     "Default": {
   4   │       "mods": null,
   5   │       "name": "Default",
   6   │       "required_targets": null
   7   │     }
   8   │   },
   9   │   "selected_profile": "",
  10   │   "version": 0
  11   │ }
───────┴────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
```

I am running sway (wlroots/Wayland) on ArchLinux with steam running on a 2nd account that has access to my XWayland and Pulse/PipeWire servers.

--- a/backend/utils/display.go
+++ b/backend/utils/display.go
@@ -16,7 +16,7 @@ func GetDisplayBounds() []image.Rectangle {
 		bounds = append(bounds, screenshot.GetDisplayBounds(i))
 	}
 
-	if runtime.GOOS == "linux" {
+	if runtime.GOOS == "linux" && n > 0 {
 		// gdk_monitor_get_geometry considers 0,0 to be the corner of the bounding box of all the monitors,
 		// not the 0,0 of the main monitor
 		boundingBox := bounds[0]
@@ -36,7 +36,10 @@ func GetDisplayBoundsAt(x, y int) image.Rectangle {
 
 	displays := GetDisplayBounds()
 
-	curDisplay := displays[0] // use main display as fallback
+	curDisplay := image.Rect(0, 0, 0, 0)
+	if len(displays) > 0 {
+	   curDisplay = displays[0] // use main display as fallback
+	}
 
 	for _, d := range displays {
 		if point.In(d) {

--- a/backend/utils/display.go
+++ b/backend/utils/display.go
@@ -38,7 +38,7 @@ func GetDisplayBoundsAt(x, y int) image.Rectangle {
 
 	curDisplay := image.Rect(0, 0, 0, 0)
 	if len(displays) > 0 {
-	   curDisplay = displays[0] // use main display as fallback
+		curDisplay = displays[0] // use main display as fallback
 	}
 
 	for _, d := range displays {

